### PR TITLE
[pyamqp] os error add

### PR DESCRIPTION
**Pull request by l0lawrence**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py [hunk 1]
```diff
@@ -530,6 +530,9 @@ async def _read(self, n, buffer=None, **kwargs):  # pylint: disable=unused-argum
             return view
         except asyncio.TimeoutError as te:
             raise ConnectionError('recv timed out (%s)' % te)
+        except OSError:
+            await self.session.close()
+            raise ConnectionError('Client Session Closed')
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py:535]: Could you please include underlying error in the raise one? And I think the phrase "client session" might be confusing.
e.g.:
```python
except OSError as e:
    await self.session.close()
    raise ConnectionError('Websocket connection closed: %r' % e) from e
```

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py [hunk 2]
```diff
@@ -504,9 +504,9 @@ async def connect(self):
             raise ValueError(
                 "Please install aiohttp library to use websocket transport."
             )
-        except OSError:
+        except OSError as e:
             await self.session.close()
-            raise ConnectionError('Client Session Closed')
+            raise ConnectionError('Websocket connection closed: %r' % e) from e
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py:509]: wondering if we need to do `% e` in the error message if we're already raising `from e`?

File: sdk/eventhub/azure-eventhub/tests/pyamqp_tests/unittest/test_websocket_exception.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import pytest
+import asyncio
+from unittest.mock import patch, MagicMock
+
+import aiohttp
+from azure.eventhub._pyamqp.aio._transport_async import WebSocketTransportAsync
+
+
+# class WebsocketException(unittest.TestCase):
+async def test_websocket_aiohttp_exception():
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/tests/pyamqp_tests/unittest/test_websocket_exception.py:10]: do we want to add a sync mock websocket test too?

Comment (l0lawrence) [reply sdk/eventhub/azure-eventhub/tests/pyamqp_tests/unittest/test_websocket_exception.py:10]: For websocket timeout exception? We can yup yup let me add it

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/tests/pyamqp_tests/unittest/test_websocket_exception.py:10]: ah maybe not then

Comment (l0lawrence) [reply sdk/eventhub/azure-eventhub/tests/pyamqp_tests/unittest/test_websocket_exception.py:10]: We except on WebSocketTimeoutException which is thrown by websocket

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py [hunk 4]
```diff
@@ -557,7 +557,10 @@ async def _read(self, n, buffer=None, **kwargs):  # pylint: disable=unused-argum
                     n = 0
             return view
         except asyncio.TimeoutError as te:
-            raise ConnectionError('Receive timed out (%s)' % te)
+            raise ConnectionError('recv timed out (%s)' % te)
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py:560]: curious why we're changing the message here?

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py:560]: (Anna mentioned in one of my PRs that we want to capitalize the first letter of the error message, so I changed this in on eof my PRs)

**Review by swathipil — APPROVED** [review]
just one comment to revert the error messages back so that the first word is capitalized (like in the feature branch).

O/w lgtm!

**Pull request merged**